### PR TITLE
Remember whether widget drawer was hidden per-room

### PR DESCRIPTION
**Pull request by anoadragon453**
Fixes vector-im/riot-web#4850

Signed-off-by: Andrew (anoa) <anoa@openmailbox.org>

### CODE REVIEW

File: src/components/structures/RoomView.js [hunk 1]
```diff
@@ -305,6 +305,20 @@ module.exports = React.createClass({
...
+        // Check if user has prompted to close this app before
+        // If so, do not show apps
+        let showWidget = localStorage.getItem(
+            room.roomId + "_show_widget_drawer");
+
+        console.warn(room);
+        console.warn("Key is: " + room.roomId + "_show_widget_drawer");
+        console.warn("showWidget is: " + showWidget);
+
+        if (showWidget == "false") {
```
Comment (rxl881) [inline src/components/structures/RoomView.js]: Should be 'showWidget === "false"', to avoid falsy comparison of missing key.

**Comment by rxl881** [comment]
Thanks for this @anoadragon453 -- This will be a really useful addition. Just a couple of comments. Can you please switch this round such that we are checking for 'hideWidgetDrawer === "true"', rather than 'showWidget == "false"'? The default state should be to show the widget drawer, and this is just adding the ability to suppress it. Hopefully this change will make it a bit clearer. Thanks!

File: src/components/structures/RoomView.js [hunk 2]
```diff
@@ -305,6 +305,20 @@ module.exports = React.createClass({
     _shouldShowApps: function(room) {
         if (!BROWSER_SUPPORTS_SANDBOX) return false;
 
+        // Check if user has prompted to close this app before
```
Comment (rxl881) [inline src/components/structures/RoomView.js]: Can you please modify this comment to:

 > // Check if user has previously chosen to hide the app drawer for this room

**Comment by anoadragon453** [comment]
Thank you guys for some awesome chat software!

**Comment by anoadragon453** [comment]
This all good to go? Eager to use on riot.im at some point :)

**Comment by t3chguy** [comment]
@dbkr / @rxl881

**Comment by ara4n** [comment]
i don’t know why this hasn’t been merged - @rxl881?

**Comment by rxl881** [comment]
Sorry that this has been delayed @t3chguy. I was waiting for @dbkr to complete/approve his review.
I'll make sure that this is merged on Monday, at the latest.

**Comment by t3chguy** [comment]
/me was just poking for @anoadragon453, not for self

**Comment by anoadragon453** [comment]
@t3chguy Thank you for that!

**Comment by rxl881** [comment]
Apologies for the delay in getting this merged @anoadragon453. It has now been merged and should be live on /develop shortly. Thanks again for the contribution!

**Comment by anoadragon453** [comment]
No worries, and thank you!

**Pull request merged**